Removed unused configuration parameters, added dispatcher and agent p… (#769)
radixdlt/babylon-gateway: sdk/typescript/README.md

const { status, transaction, stream, state } = gatewayApi
```

## Connecting to a secured gateway using authorization header

> This example shows how to provide basic auth, JWT and Bearer Token can be done the same way by passing the authorization header.

Reviewer: This sentence doesn't work grammatically.
```suggestion
This example shows how to add an authorization header to a request, in order to make requests to a secured Gateway requiring basic auth, a JWT or a Bearer Token.
```

Author: fixed

To work with this set-up you will need to:

> * Ensure your `basePath` is set up to connect to the correct host and port, on which Gateway API. is exposed.

Reviewer: ```suggestion
* Ensure your `basePath` is set up to connect to the correct host and port, on which Gateway API is exposed.
```

Author: fixed

* Set the `Authorization` header on your request, configured with your basic auth credentials for the Gateway API.

> * In the `agent` / `dispatcher` configuration, include the self-signed certificate by using the `ca` parameter, or if you understand the implications and have precautions against MITM, use `rejectUnauthorized: false` with the https agent.

Reviewer: ```suggestion
* If the Gateway server doesn't provide a valid HTTPs certificate, you will need to work around that. In the `agent` / `dispatcher` configuration, include the self-signed certificate by using the `ca` parameter, or if you understand the implications and have precautions against MITM, use `rejectUnauthorized: false` with the https agent.
```

Author: fixed

### With `node-fetch`

> An example, with the `rejectUnauthorized: false` line, assuming you are EG on localhost, or a private network where MITM attacks are mitigated.

Reviewer: I would get rid of this line. Generally Gateways will have valid certificates (unlike nodes)

Author: fixed

```typescript
import https from "node:https";
import fetch from "node-fetch"
import https from "node:https";
import { GatewayApiClient, RadixNetwork } from '@radixdlt/babylon-gateway-api-sdk'

const basicAuthUsername = "????";
const basicAuthPassword = "????"; // From your gateway set-up - provide this securely to your application

const gatewayApiClient = GatewayApiClient.initialize({
  networkId: RadixNetwork.Mainnet,
  applicationName: 'Your dApp Name',
  applicationVersion: '1.0.0',
  applicationDappDefinitionAddress: 'account_rdx12y4l35lh2543nfa9pyyzvsh64ssu0dv6fq20gg8suslwmjvkylejgj',
  agent: new https.Agent({
    keepAlive: true,

>     // NOTE - Only add the below line if you've taken precautions to avoid MITM attacks between you and the gateway api
>     rejectUnauthorized: false,

Reviewer: We shouldn't have `rejectUnauthorized: false,` in our default configuration - unlike the node, Gateways aren't expected to use self-signed certificates - and this is a scary default.

```suggestion
    // If the Gateway presents an invalid certificate, you can work around it here.
    // e.g. by using the `ca` parameter to provide a custom certificate,
    // or using `rejectUnauthorized: false` to ignore the certificate check 
    // - as long you have taken precautions to prevent a man in the middle attack.
```

  }),
  headers: {
      "Authorization": `Basic ${Buffer.from(`${basicAuthUsername}:${basicAuthPassword}`).toString("base64")}`
  }
});
```

### With native Node.JS `fetch`

If wanting to customise the certificate settings on the request, you will need install `undici` (as per [this comment](https://github.com/nodejs/undici/issues/1489#issuecomment-1543856261)):

> The below includes the `rejectUnauthorized: false` line, assuming you are EG on localhost, or a private network where MITM attacks are mitigated.

Reviewer: As above, let's get rid of this line too.

Author: fixed

```typescript
import { Agent } from 'undici';
import { GatewayApiClient, RadixNetwork } from '@radixdlt/babylon-gateway-api-sdk'

const basicAuthUsername = "????";
const basicAuthPassword = "????"; // From your gateway set-up - provide this securely to your application

const gatewayApiClient = GatewayApiClient.initialize({
  networkId: RadixNetwork.Mainnet,
  applicationName: 'Your dApp Name',
  applicationVersion: '1.0.0',
  applicationDappDefinitionAddress: 'account_rdx12y4l35lh2543nfa9pyyzvsh64ssu0dv6fq20gg8suslwmjvkylejgj',
  dispatcher: new Agent({
      connect: {
          // NOTE - Only add the below line if you've taken precautions to avoid MITM attacks between you and the gateway api

>           rejectUnauthorized: false,

Reviewer: And let's comment out this.

Author: fixed

const { status, transaction, stream, state } = gatewayApi
```

## Connecting to a secured gateway using authorization header

This example shows how to add an authorization header to a request, in order to make requests to a secured Gateway requiring basic auth, a JWT or a Bearer Token.

To work with this set-up you will need to:
* Ensure your `basePath` is set up to connect to the correct host and port, on which Gateway API is exposed.
* Set the `Authorization` header on your request, configured with your basic auth credentials for the Gateway API.

> *  If the Gateway server doesn't provide a valid HTTPs certificate, you will need to work around that. In the `agent` / `dispatcher` configuration, include the self-signed certificate by using the `ca` parameter, or if you understand the implications and have precautions against MITM, use `rejectUnauthorized: false` with the https agent.`rejectUnauthorized: false` with the https agent.

Reviewer: Last bit is duplicated
```suggestion
*  If the Gateway server doesn't provide a valid HTTPs certificate, you will need to work around that. In the `agent` / `dispatcher` configuration, include the self-signed certificate by using the `ca` parameter, or if you understand the implications and have precautions against MITM, use `rejectUnauthorized: false` with the https agent.
```
